base/int.jl

```diff
@@ -142,62 +132,37 @@ cld{T<:Integer }(x::T, y::T) = div(x,y)+(!signbit(x$y)&(rem(x,y)!=0))
 
 ## integer bitwise operations ##
 
-for T in IntTypes
-    @eval begin
-        ~(x::$T) = box($T,not_int(unbox($T,x)))
+(~){T<:UnionIntTypes}(x::T)       = box(T,not_int(unbox(T,x)))
+(&){T<:UnionIntTypes}(x::T, y::T) = box(T,and_int(unbox(T,x),unbox(T,y)))
+(|){T<:UnionIntTypes}(x::T, y::T) = box(T, or_int(unbox(T,x),unbox(T,y)))
+($){T<:UnionIntTypes}(x::T, y::T) = box(T,xor_int(unbox(T,x),unbox(T,y)))
 
-        (&)(x::$T, y::$T) = box($T,and_int(unbox($T,x),unbox($T,y)))
-        (|)(x::$T, y::$T) = box($T, or_int(unbox($T,x),unbox($T,y)))
-        ($)(x::$T, y::$T) = box($T,xor_int(unbox($T,x),unbox($T,y)))
-    end
-    for S in IntTypes
-        (S === Int128 || S === UInt128) && continue
-        @eval begin
-            <<(x::$T,  y::$S) = box($T, shl_int(unbox($T,x),unbox($S,y)))
-            >>>(x::$T, y::$S) = box($T,lshr_int(unbox($T,x),unbox($S,y)))
-        end
-        if issubtype(T,Unsigned)
-            @eval >>(x::$T, y::$S) = box($T,lshr_int(unbox($T,x),unbox($S,y)))
-        else
-            @eval >>(x::$T, y::$S) = box($T,ashr_int(unbox($T,x),unbox($S,y)))
-        end
-    end
+for T in IntTypes, S in tuple(Signed64Types..., Unsigned64Types...)
+    @eval <<(x::$T,  y::$S) = box($T, shl_int(unbox($T,x),unbox($S,y)))
```

Comment:
Simply because I hadn't had more time :grinning: to clean that up.
Also because in my earlier attempts to clean some of those up, I ran into segment violations, which I've now finally figured out how to get around (had to do with the fact that those operators, `<` and `<<` are used in this file, and need to be defined early on)